src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNull.java

```diff
@@ -74,83 +80,95 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                         return EQUALS_STRING.matches(methodInvocation) ||
-                                (EQUALS_OBJECT.matches(methodInvocation) && TypeUtils.isString(methodInvocation.getArguments().get(0).getType()))||
+                                (EQUALS_OBJECT.matches(methodInvocation) &&
+                                        TypeUtils.isString(methodInvocation.getArguments().get(0).getType())) ||
                                 EQUALS_IGNORE_CASE.matches(methodInvocation) ||
                                 CONTENT_EQUALS.matches(methodInvocation);
                     }
 
                     private void maybeHandleParentBinary(J.MethodInvocation m, final Tree parent) {
-                        if (parent instanceof J.Binary) {
-                            if (((J.Binary) parent).getOperator() == J.Binary.Type.And &&
-                                    ((J.Binary) parent).getLeft() instanceof J.Binary) {
-                                J.Binary potentialNullCheck = (J.Binary) ((J.Binary) parent).getLeft();
-                                if (isNullLiteral(potentialNullCheck.getLeft()) &&
-                                        matchesSelect(potentialNullCheck.getRight(), requireNonNull(m.getSelect())) ||
-                                    isNullLiteral(potentialNullCheck.getRight()) &&
-                                            matchesSelect(potentialNullCheck.getLeft(), requireNonNull(m.getSelect()))) {
-                                    doAfterVisit(new JavaVisitor<ExecutionContext>() {
-
-                                        private final J.Binary scope = (J.Binary) parent;
-                                        private boolean done;
-
-                                        @Override
-                                        public @Nullable J visit(@Nullable Tree tree, ExecutionContext ctx) {
-                                            return done ? (J) tree : super.visit(tree, ctx);
-                                        }
-
-                                        @Override
-                                        public J visitBinary(J.Binary binary, ExecutionContext ctx) {
-                                            if (scope.isScope(binary)) {
-                                                done = true;
-                                                return binary.getRight().withPrefix(binary.getPrefix());
-                                            }
-                                            return super.visitBinary(binary, ctx);
-                                        }
-                                    });
+                        if (parent instanceof J.Binary &&
+                                ((J.Binary) parent).getOperator() == J.Binary.Type.And &&
+                                ((J.Binary) parent).getLeft() instanceof J.Binary) {
+                            potentialNullCheck(m, (J.Binary) parent, (J.Binary) ((J.Binary) parent).getLeft());
```

Comment:
here we know whats going on `potentialNullCheck` reducing the mental load into smaller scopes making it easy/possible to digest.